src/main/java/uk/ac/cam/cl/dtg/isaac/api/AssignmentFacade.java

```diff
@@ -939,105 +941,152 @@ public Response getAssignedGroupsByGameboards(@Context final HttpServletRequest
+                    eventDetails.put(Constants.GAMEBOARD_ID_FKEY, assignmentWithID.getGameboardId());
+                    eventDetails.put(GROUP_FK, assignmentWithID.getGroupId());
+                    eventDetails.put(ASSIGNMENT_FK, assignmentWithID.getId());
+                    eventDetails.put(ASSIGNMENT_DUEDATE_FK, assignmentWithID.getDueDate());
+                    this.getLogManager().logEvent(currentlyLoggedInUser, request, IsaacServerLogType.SET_NEW_ASSIGNMENT, eventDetails);
+
+                    this.userBadgeManager.updateBadge(currentlyLoggedInUser,
+                            UserBadgeManager.Badge.TEACHER_ASSIGNMENTS_SET, assignmentWithID.getId().toString());
+
+                    tagsLoop:
+                    for (String tag : bookTags) {
+                        for (GameboardItem item : gameboard.getContents()) {
+                            if (item.getTags().contains(tag)) {
+                                this.userBadgeManager.updateBadge(currentlyLoggedInUser,
+                                        UserBadgeManager.Badge.TEACHER_BOOK_PAGES_SET, assignmentWithID.getId().toString());
+                                break tagsLoop;
+                            }
+                        }
                     }
+                    // Assigning to this group was a success
+                    assigmentSuccessfulGroupIds.add(assignmentWithID.getGroupId());
+                } catch (DuplicateAssignmentException e) {
+                    assigmentErorrs.add(new AssignmentErrorDTO(assignmentDTO.getGroupId(), e.getMessage()));
+                } catch (SegueDatabaseException e) {
+                    log.error("Database error while trying to assign work", e);
+                    assigmentErorrs.add(new AssignmentErrorDTO(assignmentDTO.getGroupId(), "Unknown database error."));
                 }
             }
-
-            return Response.ok(assignmentDTOFromClient).build();
+            return Response.ok(new AssignmentSettingResponseDTO(assigmentSuccessfulGroupIds, assigmentErorrs)).build();
         } catch (NoUserLoggedInException e) {
             return SegueErrorResponse.getNotLoggedInResponse();
-        } catch (DuplicateAssignmentException e) {
-            return new SegueErrorResponse(Status.BAD_REQUEST, e.getMessage()).toResponse();
-        } catch (SegueDatabaseException e) {
-            log.error("Database error while trying to assign work", e);
-            return new SegueErrorResponse(Status.INTERNAL_SERVER_ERROR, "Unknown database error.").toResponse();
         }
     }
 
+    /**
+     * Allows a user to assign a gameboard to group of users.
+     *
+     * @param request
+     *            - so that we can identify the current user.
+     * @param assignmentDTOFromClient a partially completed DTO for the assignment.
+     * @return an AssignmentSettingResponseDTO (see assignGameBoards)
+     */
+    @POST
+    @Path("/assign/")
```

Comment:
If I understand this correctly, this endpoint is going to go in a few releases time, right? So, should we just mark this as deprecated so we remember to remove it in the next release or something like that?

Reply:
I don't see why we would deprecate this? Especially since this single-assignment request will still be the majority of cases! There's no additional maintenance burden in maintaining it or anything, so why remove it?

Reply:
In the frontend, setting a single assignment sends a singleton list to the multiple assignment endpoint, so nowhere in the frontend actually sends a single-assignment request. The only reason I can think of keeping this endpoint is so the API is more flexible, but the frontend definitely isn't using this flexibility at the moment.

Reply:
I feel like it's best to deprecate code whose functionality is subsumed by other code, just so that there's less redundant stuff when browsing the codebase.

Reply:
That's what I meant, yes. I couldn't find any mention of `/assign/` in the front-end and, given the singleton list use case, I thought deprecating the old endpoint reduced maintenance and attack surface.

Reply:
Okay, fine. I think it's no longer actually the same endpoint anyway, since the value it returns is now very different! It's a breaking change.
Let's leave it marked this way.